pkg/ring/lifecycler.go

```diff
@@ -391,23 +404,41 @@ func (i *Lifecycler) ZonesCount() int {
 	return i.zonesCount
 }
 
+func (i *Lifecycler) Join() {
+	select {
+	case i.autojoinChan <- struct{}{}:
+	default:
```

Comment:
What do you think about logging here, in case sending to `i.autojoinChan` is rejected?